python/tests/test_umap.py

```diff
@@ -0,0 +1,165 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import math
+from typing import Tuple
+
+import cupy as cp
+import numpy as np
+import pytest
+from cuml.metrics import trustworthiness
+from sklearn.datasets import load_digits, load_iris
+
+from .sparksession import CleanSparkSession
+from .utils import create_pyspark_dataframe, cuml_supported_data_types
+
+
+def _load_dataset(dataset: str, n_rows: int) -> Tuple[np.ndarray, np.ndarray]:
+    if dataset == "digits":
+        local_X, local_y = load_digits(return_X_y=True)
+
+    else:  # dataset == "iris"
+        local_X, local_y = load_iris(return_X_y=True)
+
+    local_X = cp.asarray(local_X)
+    local_y = cp.asarray(local_y)
+
+    local_X = local_X.repeat(math.ceil(n_rows / len(local_X)), axis=0)
+    local_y = local_y.repeat(math.ceil(n_rows / len(local_y)), axis=0)
+
+    # Add some gaussian noise
+    local_X += cp.random.standard_normal(local_X.shape, dtype=cp.float32)
+
+    return local_X, local_y
+
+
+def _local_umap_trustworthiness(
+    local_X: np.ndarray,
+    local_y: np.ndarray,
+    n_neighbors: int,
+    supervised: bool,
+) -> float:
+    from cuml.manifold import UMAP
+
+    local_model = UMAP(n_neighbors=n_neighbors, random_state=42, init="random")
+    y_train = local_y if supervised else None
+    local_model.fit(local_X, y=y_train)
+
+    embedding = local_model.embedding_
+    return trustworthiness(local_X, embedding, n_neighbors=n_neighbors, batch_size=5000)
+
+
+def _spark_umap_trustworthiness(
+    local_X: np.ndarray,
+    local_y: np.ndarray,
+    n_neighbors: int,
+    supervised: bool,
+    n_parts: int,
+    sampling_ratio: float,
+    dtype: np.dtype,
+) -> float:
+    from spark_rapids_ml.umap import UMAP
+
+    local_model = UMAP(
+        n_neighbors=n_neighbors,
+        sample_fraction=sampling_ratio,
+        random_state=42,
+        init="random",
+        num_workers=n_parts,
+    )
+
+    with CleanSparkSession() as spark:
+        if supervised:
+            data_df, features_col, label_col = create_pyspark_dataframe(
+                spark, "array", dtype, local_X, local_y
+            )
+            assert label_col is not None
+            local_model.setLabelCol(label_col)
+        else:
+            data_df, features_col, _ = create_pyspark_dataframe(
+                spark, "array", dtype, local_X, None
```

Comment:
Since there is only 1 test, I'd like to suggest that we'd better cover more different spark types, like vector/columnar input